scripts/world_bank/worldbank.py

```diff
@@ -75,52 +83,65 @@ def read_worldbank(iso3166alpha3):
             Takes approximately 10 seconds to download and
             tidy one country in a Jupyter notebook.
     """
-    country_zip = ("http://api.worldbank.org/v2/en/country/" + iso3166alpha3 +
-                   "?downloadformat=csv")
-    r = requests.get(country_zip)
-    filebytes = io.BytesIO(r.content)
-    myzipfile = zipfile.ZipFile(filebytes)
-
-    # We need to select the data file which starts with "API",
-    # but does not have an otherwise regular filename structure.
-    file_to_open = None
-    for file in myzipfile.namelist():
-        if file.startswith("API"):
-            file_to_open = file
-            break
-    assert file_to_open is not None, \
-        "Failed to find data for" + iso3166alpha3
-
-    df = None
-    # Captures any text contained in double quotatations.
-    line_match = re.compile(r"\"([^\"]*)\"")
-
-    for line in myzipfile.open(file_to_open).readlines():
-        # Cells are contained in quotations and comma separated.
-        cols = line_match.findall(line.decode("utf-8"))
-
-        # CSVs include header informational lines which should be ignored.
-        if len(cols) > 2:
-            # Use first row as the header.
-            if df is None:
-                df = pd.DataFrame(columns=cols)
-            else:
-                df = df.append(pd.DataFrame([cols], columns=df.columns),
-                               ignore_index=True)
-
-    df = df.rename(columns=WORLDBANK_COL_REMAP)
-
-    # Turn each year into its own row.
-    df = df.set_index(
-        ['CountryName', 'CountryCode', 'IndicatorName', 'IndicatorCode'])
-    df = df.stack()
-    df.index = df.index.rename('year', level=4)
-    df.name = "Value"
-    df = df.reset_index()
-
-    # Convert to numeric and drop empty values.
-    df['Value'] = pd.to_numeric(df['Value'])
-    df = df.dropna()
+    if fetchFromSource:
+        print(f"Downloading {iso3166alpha3}")
+        country_zip = ("http://api.worldbank.org/v2/en/country/" +
+                       iso3166alpha3 + "?downloadformat=csv")
+        r = requests.get(country_zip)
```

Comment:
if this is in general needed, can use: https://pypi.org/project/retry/

Reply:
Nice, thanks!